src/components/PostDetails.tsx

```diff
@@ -1,106 +1,124 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { createComment, deleteComment, getPostComments } from './api/users';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorComments, setIsErrorComments] = useState(false);
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const isNoComments =
+    selectedPost && comments.length === 0 && !isLoading && !isErrorComments;
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsShownForm(false);
+    if (selectedPost) {
+      setIsLoading(true);
+      getPostComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsErrorComments(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addComment = ({ name, email, body, postId }: Omit<Comment, 'id'>) => {
+    return createComment({ name, email, body, postId }).then(newComment =>
+      setComments(currentComments => [...currentComments, newComment]),
+    );
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    setComments(currentComments =>
+      currentComments.filter(comment => comment.id !== commentId),
+    );
+    deleteComment(commentId);
```

Comment:
You are deleting a comment from the state before the request to delete it on the server has been completed. This might lead to a situation where the comment is deleted in the UI, but the server deletion fails. A better approach would be to wait for the server deletion to complete and then delete the comment from the state.